libraries/script-engine/src/ScriptEngine.cpp

```diff
@@ -1875,7 +1876,7 @@ QScriptValue ScriptEngine::require(const QString& moduleId) {
     //   `delete Script.require.cache[Script.require.resolve(moduleId)];`
 
     // cacheMeta is just used right now to tell deleted keys apart from undefined ones
-    bool invalidateCache = module.isUndefined() && cacheMeta.property(moduleId).isValid();
+    bool invalidateCache = (module.isUndefined() && cacheMeta.property(moduleId).isValid()) || forceRedownload;
```

Comment:
```suggestion
    bool invalidateCache = forceRedownload || (module.isUndefined() && cacheMeta.property(moduleId).isValid());
```
this way the code can short circuit and save us some work